pkg/asset/cluster/tfvars.go

```diff
@@ -407,6 +409,22 @@ func (t *TerraformVariables) Generate(parents asset.Parents) error {
 			ServiceAccount:   string(sess.Credentials.JSON),
 		}
 
+		credsProject := installConfig.Config.GCP.ProjectID
+		if installConfig.Config.GCP.NetworkProjectID != "" {
+			credsProject = installConfig.Config.GCP.NetworkProjectID
+		}
+		client, err := gcpconfig.NewClient(context.Background())
+		if err != nil {
+			return err
+		}
+		permissions, err := client.GetProjectPermissions(context.Background(), credsProject, []string{
+			tfVarsGCPFirewallPermission,
+		})
+		if err != nil {
+			return err
+		}
+		createFirewallRules := permissions.Has(tfVarsGCPFirewallPermission)
```

Comment:
We should only be doing this check (and only disabling createFirewallRules) for XPN installs. 

It would be good to explain that we're doing it because the user may not have permissions in the host project... either in a comment here or in the commit message